hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -8441,9 +8446,9 @@ public void startRegionOperation(Operation op) throws IOException {
     // Update regionLockHolders ONLY for any startRegionOperation call that is invoked from
     // an RPC handler
     Thread thisThread = Thread.currentThread();
-    if (isInterruptableOp) {
```

Comment:
if this is changing then the comment on line 8441-8442 should also change (we are no longer updating `regionLockHolders` ONLY for calls invoked from RPC handlers, but also for internal operations like REPLAY, or COMPACT_REGION, or SNAPSHOT, or ANY (used in several places, like batch mutate internal or refresh store files.)

The comment should reflect the code. Similarly for the comment at the declaration of `regionLockHolders` for the class. This assumes that the logic change is okay, of course.

I do worry about the logic change, because of calls to `enableInterrupts` and `disableInterrupts`. That will change the value of this flag in a way that was not possible before (because it was not present before, and therefore `computeIfPresent` would not do anything.)

Reply:
`disableInterrupts()` and `enableInterrupts()` are only meant for the mutation RPCs: interruption is disabled right before making WAL edit until the batchMutation is completed. Hence, only mutation RPCs will go through this cycle.
Hence, as far as other non-interruptible threads are concerned, even if they do stay in the same map (like earlier version), we are not going to enable interruption for them. Similarly, any non-mutation interruptible threads are not going to be converted to non-interruptible by any workflow. So, single map was sufficient IMO.

I agree with David above that the comment should reflect the new code changes. But having two maps is becoming bit difficult to understand, more so when one looks back at the history of which change had to introduce two maps :)